src/engine/bank05.asm

```diff
@@ -4196,66 +4464,1748 @@ Func_16695: ; 16695 (5:6695)
+	ret
+
+; tests for the following conditions:
+; - player is under No Damage substatus;
+; - second move is unusable;
+; - second move deals no damage;
+; if any are true, returns score of $80 + 5.
+HandleSwordsDanceAndFocusEnergy: ; 16ecb (5:6ecb)
+	ld a, [wAICannotDamage]
+	or a
+	jr nz, .success
+	ld a, $01 ; second move
+	ld [wSelectedMoveIndex], a
+	call CheckIfSelectedMoveIsUnusable
+	jr c, .success
+	ld a, $01 ; second move
+	call EstimateDamage_VersusDefendingCard
+	ld a, [wDamage]
+	or a
+	jp nz, HandleSpecialAIMoves.zero
+.success
+	ld a, $85
+	ret
+
+; checks player's active card color, then
+; loops through bench looking for a Pokémon
+; with that same color.
+; if none are found, returns score of $80 + 2.
+HandleElectrode2ChainLightning: ; 16eea (5:6eea)
+	call SwapTurn
+	call GetArenaCardColor
+	call SwapTurn
+	ld b, a
+	ld a, DUELVARS_BENCH
+	call GetTurnDuelistVariable
+.loop
+	ld a, [hli]
+	cp $ff
+	jr z, .success
+	push bc
+	call GetCardIDFromDeckIndex
+	call GetCardType
+	pop bc
+	cp b
+	jr nz, .loop
+	jp HandleSpecialAIMoves.zero
+.success
+	ld a, $82
+	ret
+
+HandleMew3DevolutionBeam: ; 16f0f (5:6f0f)
+	call LookForCardThatIsKnockedOutOnDevolution
+	jp nc, HandleSpecialAIMoves.zero
+	ld a, $85
+	ret
+
+; first checks if card is confused, and if so return 0.
+; then checks number of Pokémon in bench that are viable to use:
+; - if that number is < 2  and this move is Conversion 1 OR
+; - if that number is >= 2 and this move is Conversion 2
+; then return score of $80 + 2.
+; otherwise return score of $80 + 1.
+HandlePorygonConversion: ; 16f18 (5:6f18)
+	ld a, DUELVARS_ARENA_CARD_STATUS
+	call GetTurnDuelistVariable
+	and CNF_SLP_PRZ
+	cp CONFUSED
+	jp z, HandleSpecialAIMoves.zero
+
+	ld a, [wSelectedMoveIndex]
+	or a
+	jr nz, .conversion_2
+
+; conversion 1
+	call CheckIfBenchCardsAreAtHalfHPCanEvolveAndUseSecondMove
+	cp 2
+	jr c, .low_score
+	ld a, $82
+	ret
+
+.conversion_2
+	call CheckIfBenchCardsAreAtHalfHPCanEvolveAndUseSecondMove
+	cp 2
+	jr nc, .low_score
+	ld a, $82
+	ret
+
+.low_score
+	ld a, $81
+	ret
+
+; if any Psychic Energy is found in the Discard Pile,
+; return a score of $80 + 2.
+HandleEnergyAbsorption: ; 16f41 (5:6f41)
+	ld e, PSYCHIC_ENERGY
+	ld a, CARD_LOCATION_DISCARD_PILE
+	call CheckIfAnyCardIDinLocation
+	jp nc, HandleSpecialAIMoves.zero
+	ld a, $82
+	ret
+
+; if player has cards in hand, AI calls Random:
+; - 1/3 chance to encourage move regardless;
+; - 1/3 chance to dismiss move regardless;
+; - 1/3 change to make some checks to player's hand.
+; AI tallies number of basic cards in hand, and if this
+; number is >= 2, encourage move.
+; otherwise, if it finds an evolution card in hand that
+; can evolve a card in player's deck, encourage.
+; if encouraged, returns a score of $80 + 3.
+HandleNinetalesMixUp: ; 16f4e (5:6f4e)
+	ld a, DUELVARS_NUMBER_OF_CARDS_IN_HAND
+	call GetNonTurnDuelistVariable
+	or a
+	ret z ; return if no hand cards
+
+	ld a, 3
+	call Random
+	or a
+	jr z, .encourage ; if a = 0
```

Comment:
this and the comment a bit below seem a little superfluous. They could possibly be removed or replaced with "if random = 0", etc?